src/signals/incident-management/containers/Dashboard/components/Filter/Filter.test.tsx

```diff
@@ -0,0 +1,247 @@
+    )
+
+    expect(
+      screen.queryByRole('option', { name: 'Amsterdamse Bos' })
+    ).not.toBeInTheDocument()
+  })
+
+  it('should select a department and thus change selectable categories', async () => {
+    render(<Filter callback={mockCallback} />)
+
+    expect(
+      screen.getByRole('listbox', {
+        name: 'Actie Service Centr...',
+      })
+    ).toBeInTheDocument()
+
+    userEvent.click(
+      screen.getByRole('listbox', {
+        name: 'Categorie',
+      })
+    )
+
+    expect(
+      screen.queryByRole('option', { name: 'Bedrijfsafval' })
+    ).toBeInTheDocument()
+
+    userEvent.click(
+      screen.getByRole('listbox', {
+        name: 'Actie Service Centr...',
+      })
+    )
+
+    userEvent.click(
+      screen.getByRole('option', {
+        name: 'Politie',
+      })
+    )
+
+    expect(
+      screen.queryByRole('listbox', { name: 'Bedrijfsafval' })
+    ).not.toBeInTheDocument()
+
+    expect(
+      screen.queryByRole('listbox', { name: 'Categorie' })
+    ).toBeInTheDocument()
+  })
+
+  it('should select a department and reset form by using keyboard', () => {
+    render(<Filter callback={mockCallback} />)
+
+    expect(
+      screen.getByRole('listbox', {
+        name: 'Actie Service Centr...',
+      })
+    ).toBeInTheDocument()
+
+    screen
+      .getByRole('listbox', {
+        name: 'Categorie',
+      })
+      .focus()
+
+    act(() => {
+      fireEvent.keyDown(
+        screen.getByRole('listbox', {
+          name: 'Categorie',
+        }),
+        { key: 'Enter' }
+      )
+    })
+
+    screen
+      .getByRole('option', {
+        name: 'Bedrijfsafval',
+      })
+      .focus()
+
+    expect(
+      screen.getByRole('option', {
+        name: 'Bedrijfsafval',
+      })
+    ).toBeInTheDocument()
+
+    act(() => {
+      fireEvent.keyDown(
+        screen.getByRole('option', {
+          name: 'Bedrijfsafval',
+        }),
+        { key: 'Enter' }
+      )
+    })
+
+    expect(
+      screen.getByRole('listbox', {
+        name: 'Bedrijfsafval',
+      })
+    ).toBeInTheDocument()
+
+    screen
+      .getByRole('button', {
+        name: 'Wis filters',
+      })
+      .focus()
+
+    act(() => {
+      fireEvent.keyDown(
+        screen.getByRole('button', {
+          name: 'Wis filters',
+        }),
+        {
+          key: 'Enter',
+        }
+      )
+    })
+
+    expect(
+      screen.queryByRole('listbox', {
+        name: 'Bedrijfsafval',
+      })
+    ).not.toBeInTheDocument()
+  })
+
+  it('should select a department, a custom category and reset back and call callback each time', async () => {
+    render(<Filter callback={mockCallback} />)
+
+    userEvent.click(
+      screen.getByRole('listbox', {
+        name: 'Actie Service Centr...',
+      })
+    )
+
+    userEvent.click(
+      screen.getByRole('option', {
+        name: 'Afval en Grondstoffen',
+      })
+    )
+
+    expect(mockCallback).toBeCalledWith('department=AEG')
+
+    mockCallback.mockReset()
```

Comment:
You don't need to reset this mock. If you don't you can add `expect(mockCallback).toBeCalledTimes(3)` in the end to be sure it is called the right amount of times.